### ISSUE TITLE
allow data-f-foreach to work when nested

### ISSUE THREAD

**Issue by mmrj** [forio/flow.js#30]
in 0.8.3, 
```
                <ul data-f-foreach="regions">
                    <li></li>
                    <ul data-f-foreach="sales">
                        <li></li>
                    </ul>
                </ul>
```
prints out the attached:

![screen shot 2015-06-05 at 11 28 26 am](https://cloud.githubusercontent.com/assets/1653043/8012401/0ec356a6-0b76-11e5-892f-e46036b30a13.png)

I would expect it to print out
```
* North
  - 100
  - 400
  - 300
  - 200
* East
  - 100
  - 400
  - 300
  - 200
```
etc. 
(edit: updated expectation to match exact data)

**Comment by narenranjit** [comment]
Can you attach the html it printed out? Also what the array values were?

**Comment by mmrj** [comment]
from the .jl file:
```
regions = { "North", "East", "South", "West" }
sales = {100, 400, 300, 200}
```

from the generated html:
```
<ul data-f-foreach="regions">
  <li>North</li>
  <ul data-f-foreach="sales">North</ul>
  <li>East</li>
  <ul data-f-foreach="sales">East</ul>
  <li>South</li>
  <ul data-f-foreach="sales">South</ul>
  <li>West</li>
  <ul data-f-foreach="sales">West</ul>
</ul>
```

**Comment by mmrj** [comment]
this is with https://forio.com/tools/js-libs/flow/0.8.3/flow.js -- not sure if that's the latest or not.

**Comment by mmrj** [comment]
with 0.9.0 this still doesn't work, but now the generated html is: 
```
<ul data-f-foreach="regions">
  <li>North</li>
  <ul data-f-foreach="sales"></ul>
  <li>East</li>
  <ul data-f-foreach="sales"></ul>
  <li>South</li>
  <ul data-f-foreach="sales"></ul>
  <li>West</li>
  <ul data-f-foreach="sales"></ul>
</ul>
```

**Comment by narenranjit** [comment]
This should work now, as long as you don't use nested templates like:
```
<ul data-f-foreach="regions">
   <li data-f-foreach="sales">
     <%= value %> <!-- this will be the 'region' value, not 'sales' value -->
   </li>
</ul>
```

**Comment by mmrj** [comment]
testing using flow.js from latest dist/ folder, from https://github.com/forio/flow.js/commit/64b5eec6b960ee6b05f3cae2f5e80fe6739bd799) 

test case 1) verified works for ul and li as per above, e.g.: 

```
        <ul data-f-foreach="Outer">
            <li><%= _.last(value) %>
                <ul data-f-foreach="Inner">
                <li></li>
                </ul>
            </li>
        </ul>
```

the _.last is only so that when we step the model, we don't get repeats ... note that the same thing will NOT work inside the inner `<li></li>` because the `value` isn't scoped: you only get one. if you use it before the inner `foreach`, it's the value of the element of the outer array; if you use it within the inner `foreach`, it's the value of the element of the inner array. (probably future improvement, but outside the scope of this issue.

test case 2) i couldn't figure out a use case for nested looping of tables that used nested `foreach`es and was also valid html. 

test case 3) however, i did successfully nest `foreach`es using both a table and a list:
```
            <tr data-f-foreach="Outer"><td>
                <span><%= _.last(value) %>
                <ul data-f-foreach="Inner"><li></li></ul>
            </td></tr>
```
creates one cell for each value of Outer, displays the value of Outer, and then underneath, lists all of the values of Inner.

**Comment by mmrj** [comment]
reopening based on Epicenter platform team feedback during today's platform sprint demo (although @narenranjit, maybe based on this you'll want to (re)close this issue and open a new one?)

feedback was:
 (a) it seems incomplete that you can't reference the outer foreach variable within the html of the inner foreach variable
 (b) strong recommendation that the syntax here be more similar to the "for each" of other languages, which is one way to accomplish (a)

short term suggestion was: remove from the example on this from the documentation / don't promote usage of nested foreaches until we've resolved.

**Comment by narenranjit** [comment]
Solved by PR #79

**Comment by mmrj** [comment]
testing details in #79.
